Enyim.Caching/Memcached/MemcachedNode.cs

```diff
@@ -387,6 +401,95 @@ internal async Task InitPoolAsync()
                 }
             }
 
+            private void StartReconciliationTask()
+            {
+                if (_connectionIdleTimeout == TimeSpan.Zero)
+                    return;
+
+                var reconcileTimer = new PeriodicTimer(_connectionIdleTimeout);
+                _ = RunTimer();
+
+                async Task RunTimer()
+                {
+                    while (await reconcileTimer.WaitForNextTickAsync().ConfigureAwait(false))
+                    {
+                        try
+                        {
+                            using var source = new CancellationTokenSource(_connectionIdleTimeout);
+                            await ReconcileAsync(source.Token).ConfigureAwait(false);
+                            _metricFunctions.Set("cache_connection_count", (ulong)(maxItems - _semaphore.CurrentCount + _freeItems.Count), _endPointStr);
+                        }
+                        catch (Exception e)
+                        {
+                            _logger.LogWarning("ReconciliationTaskFailed", new EventId(0), e);
+                        }
+                    }
+                }
+
+            }
+
+            private async Task ReconcileAsync(CancellationToken cancellationToken)
+            {
+                // synchronize access to this method as only one clean routine should be run at a time
+                await _cleanSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
+                try
+                {
+                    var waitTimeout = TimeSpan.FromMilliseconds(10);
+                    while (true)
+                    {
+                       
+                            // Calculate if current connection count <= minimum pool size
+                        if (maxItems - _semaphore.CurrentCount + _freeItems.Count <= minItems)
+                                return;
+
+                        if (!await _semaphore.WaitAsync(waitTimeout, cancellationToken).ConfigureAwait(false))
+                            return;
+
+                        try
+                        {
+                            PooledSocket retval = null;
+                            lock (_freeItems)
+                            {
+                                if (_freeItems.Count > 0)
+                                {
+                                    retval = _freeItems.Last!.Value;
+                                    _freeItems.RemoveLast();
+                                }
+                            }
+
+                            if (retval is null)
+                                return;
+
+                            var idleTime = DateTime.UtcNow - retval.LastConnectionTimestamp;
+
+
+                            if (idleTime > _connectionIdleTimeout)
+                            {
+                                _logger.LogInformation("{0} pool found session {1} to clean up", ownerNode, retval.InstanceId);

```

Comment:
If possible use Aeplcore.logging, since for this, you need to enable Microsoft logging to be visible, thus after enabling that, the influx of logs is way too much

Reply:
We don’t want to enable this log for production. We will update it if needed in the future